packages/app/src/app/components/CreateSandbox/Import/SuggestedRepositories.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import formatDistanceStrict from 'date-fns/formatDistanceStrict';
+import { zonedTimeToUtc } from 'date-fns-tz';
+import styled from 'styled-components';
+import VisuallyHidden from '@reach/visually-hidden';
+
+import {
+  Stack,
+  Text,
+  Icon,
+  InteractiveOverlay,
+  Button,
+  Element,
+} from '@codesandbox/components';
+import { v2DefaultBranchUrl } from '@codesandbox/common/lib/utils/url-generator';
+
+import { useActions, useAppState } from 'app/overmind';
+import { useGitHuPermissions } from 'app/hooks/useGitHubPermissions';
+
+import { fuzzyMatchGithubToCsb } from './utils';
+import { useGithubAccounts } from './useGithubOrganizations';
+import { useGitHubAccountRepositories } from './useGitHubAccountRepositories';
+import { AccountSelect } from './AccountSelect';
+
+export const SuggestedRepositories = () => {
+  const { activeTeamInfo } = useAppState();
+  const { signInGithubClicked } = useActions();
+  const { restrictsPrivateRepos } = useGitHuPermissions();
+
+  const [selectedAccount, setSelectedAccount] = useState<string | undefined>();
+  const githubAccounts = useGithubAccounts();
+
+  const selectOptions = useMemo(
+    () =>
+      githubAccounts.data
+        ? [githubAccounts.data.personal, ...githubAccounts.data.organizations]
+        : undefined,
+    [githubAccounts.data]
+  );
+
+  useEffect(() => {
+    // Set initially selected account bazed on fuzzy matching
+    if (githubAccounts.state === 'ready' && selectedAccount === undefined) {
+      const match = fuzzyMatchGithubToCsb(activeTeamInfo?.name, selectOptions);
+
+      setSelectedAccount(match.login);
+    }
+  }, [githubAccounts.state, selectedAccount, activeTeamInfo, selectOptions]);
+
+  // eslint-disable-next-line no-nested-ternary
+  const selectedAccountType = selectedAccount
+    ? selectedAccount === githubAccounts?.data?.personal?.login
+      ? 'personal'
+      : 'organization'
+    : undefined;
+
+  const githubRepos = useGitHubAccountRepositories({
+    name: selectedAccount,
+    accountType: selectedAccountType,
+  });
+
+  return githubAccounts.state === 'ready' && selectedAccount ? (
+    <Stack
+      direction="vertical"
+      gap={3}
+      css={{ fontFamily: 'Inter', marginBottom: '16px' }}
+    >
+      <Stack justify="space-between">
+        <AccountSelect
+          options={selectOptions}
+          value={selectedAccount}
+          onChange={(account: string) => {
+            setSelectedAccount(account);
+          }}
+        />
+      </Stack>
+      {githubRepos.state === 'ready' ? (
```

Comment:
Looks like when switching between organizations, the repos from the previous org are shown while the repos from the newly selected repo get loaded.

https://user-images.githubusercontent.com/24959348/219976794-561187b6-2f97-4377-9897-459ffc8b6de7.mov

Reply:
I'll look into the overall loading state too, good catch!

Reply:
Resolved with in [dbe64c0](https://github.com/codesandbox/codesandbox-client/pull/7436/commits/dbe64c032aca741272447280166c9d22f3f13027).